docs/decisions/0029-self-service-purchasing-BFF.rst

```diff
@@ -0,0 +1,496 @@
+
+    Request:
+    {
+        "admin_email": "admin@example.com",
+        "enterprise_slug": "example-corp",
+        "quantity": 10,
+        "stripe_price_id": "price_1MoBy5LkdIwHu7ixZhnattbh"
+    }
+
+    Response (201 Created):
+    {
+        "checkout_session": {
+            "client_secret": "cs_test_1234567890abcdef",
+            "expires_at": "1751323210"
+        }
+    }
+
+    Response (422 Unprocessable Entity - Validation Failed):
+    {
+        "admin_email": {
+            "error_code": "not_registered",
+            "developer_message": "The provided email has not yet been registered."
+        },
+        "enterprise_slug": {
+            "error_code": "existing_enterprise_customer",
+            "developer_message": "Slug conflicts with existing customer."
+        }
+    }
+
+**4. Create Portal Session**
+
+.. code-block::
+
+    GET /api/v1/customer-billing/<customer_uuid>/portal-session
+
+    Authentication: JWT (required)
+    Authorization: Only allow admins for the given customer_uuid
+    Purpose: URL for any "Billing Portal" button (placed anywhere in admin portal or emails). 302 Redirect to Billing Portal URL.
+    Side-Effects: Create a new Stripe Billing Portal Session (consider rate limiting and caching).
+
+    Response (302 Redirect):
+    Location: https://billing.stripe.com/session/bps_1234567890abcdef
+
+**5. Stripe Webhook Handler**
+
+.. code-block::
+
+    POST /api/v1/customer-billing/stripe-webhook
+
+    Authentication: Payload signature validation
+    Authorization: Only allow Stripe system user
+    Purpose: Receive specific Stripe events to trigger email communications with the admin as needed
+    Side-Effects: Permanent storage of event payload in DB, Possible triggering of Celery task
+
+    Supported webhook events:
+    - checkout.session.completed
+    - invoice.paid
+    - customer.subscription.trial_will_end
+    - customer.subscription.deleted
+    - payment_method.attached
+
+    Request (from Stripe):
+    {
+        "id": "evt_1MoBy5LkdIwHu7ixZhnattbh",
+        "object": "event",
+        "type": "checkout.session.completed",
+        "data": {
+            "object": {
+                "id": "cs_test_1234567890abcdef",
+                "customer": "cus_test_customer123",
+                "subscription": "sub_test_subscription456"
+            }
+        }
+    }
+
+    Response (200 OK):
+    {
+        "received": true,
+        "event_id": "evt_1MoBy5LkdIwHu7ixZhnattbh"
+    }
+
+
+Form State Management
+---------------------
+
+Form state persistence is handled entirely on the frontend using in-memory
+state management.  Here is an example ``zustand`` state hook:
+
+.. code-block:: javascript
+
+    // Frontend state management strategy
+    import { create } from 'zustand';
+
+    const useCheckoutFormStore = create<FormStore>(
+      (set) => ({
+        formData: {
+          account: {},  // page 1 state.
+          trial: {},    // page 2 state.
```

Comment:
[inform] This is a great layout for state management data-structure but may change based on how the the frontend iterates (mostly around understanding the capabilities and nuances of zustand). This data-structure is the north star during the first iteration.